src/components/OrderEditor/hooks/useOrderEditorTPSL.ts

```diff
@@ -0,0 +1,577 @@
+  const gasLimits = useSelector(selectGasLimits);
+  const gasPrice = useSelector(selectGasPrice);
+  const tokensData = useSelector(selectTokensData);
+  const chainId = useSelector(selectChainId)!;
+  const isAutoCancelTPSL = useSelector(selectIsAutoCancelTPSLEnabled);
+  const maxAutoCancelOrders = useSelector(selectMaxAutoCancelOrders);
+  const userReferralInfo = useSelector(selectUserReferralInfo);
+  const uiFeeFactor = useSelector(selectUiFeeFactor)!;
+  const isSetAcceptablePriceImpactEnabled = useSelector(selectIsSetAcceptablePriceImpactEnabled);
+  const selectOrdersByPositionKey = useMemo(() => makeSelectOrdersByPositionKey(positionKey), [positionKey]);
+  const positionOrders = useSelector(selectOrdersByPositionKey);
+  const market = useMarketInfo(order?.marketAddress ?? "");
+  const markPrice = order ? (order.isLong ? indexToken?.prices?.minPrice : indexToken?.prices?.maxPrice) : undefined;
+  const positionIndexToken = order?.indexToken;
+
+  const [isTpSlEnabled, setIsTpSlEnabled] = useState(false);
+  const [tpPriceInputValue, setTpPriceInputValue] = useState("");
+  const [slPriceInputValue, setSlPriceInputValue] = useState("");
+
+  useEffect(() => {
+    setIsTpSlEnabled(false);
+    setTpPriceInputValue("");
+    setSlPriceInputValue("");
+  }, [order?.key]);
+
+  const isLimitOrStopIncrease = useMemo(
+    () => (order ? isLimitIncreaseOrderType(order.orderType) || isStopIncreaseOrderType(order.orderType) : false),
+    [order]
+  );
+
+  const totalSizeUsdForTpSl = useMemo(() => {
+    if (!isLimitOrStopIncrease || !order) return 0n;
+
+    const baseSize = existingPosition?.sizeInUsd ?? 0n;
+    const orderSize = sizeDeltaUsd ?? order.sizeDeltaUsd ?? 0n;
+
+    return baseSize + orderSize;
+  }, [existingPosition?.sizeInUsd, isLimitOrStopIncrease, order, sizeDeltaUsd]);
+
+  const totalSizeInTokensForTpSl = useMemo(() => {
+    if (!isLimitOrStopIncrease || !order) return 0n;
+
+    const baseSizeInTokens = existingPosition?.sizeInTokens ?? 0n;
+
+    const orderSizeInTokens =
+      increaseAmounts?.sizeDeltaInTokens ??
+      (positionIndexToken && (sizeDeltaUsd ?? order.sizeDeltaUsd)
+        ? convertToTokenAmount(
+            sizeDeltaUsd ?? order.sizeDeltaUsd,
+            positionIndexToken.decimals,
+            triggerPrice ?? order.triggerPrice
+          ) ?? 0n
+        : 0n);
+
+    return baseSizeInTokens + (orderSizeInTokens ?? 0n);
+  }, [
+    existingPosition?.sizeInTokens,
+    increaseAmounts?.sizeDeltaInTokens,
+    isLimitOrStopIncrease,
+    order,
+    positionIndexToken,
+    sizeDeltaUsd,
+    triggerPrice,
+  ]);
+
+  const collateralDeltaAmountForTpSl =
+    increaseAmounts?.collateralDeltaAmount ??
+    (isLimitOrStopIncrease && order ? order.initialCollateralDeltaAmount : undefined) ??
+    0n;
+
+  const totalCollateralAmountForTpSl = (existingPosition?.collateralAmount ?? 0n) + collateralDeltaAmountForTpSl;
```

Comment:
would be better to move all calcs to pure function(s) and just use it in hooks for better testability